TKeazirian.HTTPServer/Controller.cs

```diff
@@ -15,5 +15,13 @@ public static byte[] GenerateResponse(string request)
             var responseToSend = Encoding.ASCII.GetBytes(constructedResponse);
             return responseToSend;
         }
+
+        public static byte[] GenerateErrorResponse()
+        {
+            var errorResponse = "HTTP/1.1 500 Internal Server Error";
+
+            var errorResponseToSend = Encoding.ASCII.GetBytes(errorResponse);
+            return errorResponseToSend;
+        }
```

Comment:
The `Controller` seems to do several things.  It builds an "Echo body" response it builds an `ErrorResponse`, and it also appears to be doing all your parsing.  What is the problem you are trying to solve and is there a way to break this up further?